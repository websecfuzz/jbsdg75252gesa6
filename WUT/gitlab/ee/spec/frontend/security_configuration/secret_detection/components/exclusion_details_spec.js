import { shallowMount } from '@vue/test-utils';
import ExclusionDetails from 'ee/security_configuration/secret_detection/components/exclusion_details.vue';
import { projectSecurityExclusions } from '../mock_data';

const [exclusion] = projectSecurityExclusions;

describe('ExclusionDetails', () => {
  let wrapper;

  const createComponent = () => {
    wrapper = shallowMount(ExclusionDetails, {
      propsData: {
        exclusion,
      },
    });
  };
  beforeEach(() => {
    createComponent();
  });

  it('renders component', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('matches the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
