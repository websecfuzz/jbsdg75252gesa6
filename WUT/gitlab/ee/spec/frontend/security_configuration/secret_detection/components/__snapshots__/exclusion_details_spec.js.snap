// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExclusionDetails matches the snapshot 1`] = `
<div
  class="gl-flex gl-flex-col gl-gap-5"
>
  <detail-stub
    label="Type"
    value="Path"
  />
  <detail-stub
    label="Value"
    value="tmp"
  />
  <detail-stub
    label="Description"
    value="test1"
  />
  <detail-stub
    label="Enforcement"
  >
    <gl-icon-stub
      class="text-success"
      name="check"
      size="16"
      variant="current"
    />
    Secret push protection
  </detail-stub>
  <detail-stub
    label="Status"
  >
    <span
      class="gl-text-success"
    >
      <gl-icon-stub
        name="check-circle-filled"
        size="16"
        variant="current"
      />
      Enabled
    </span>
  </detail-stub>
  <detail-stub
    label="Created"
    value="Sep 19, 2024"
  />
  <detail-stub
    label="Updated"
    value="Sep 19, 2024"
  />
</div>
`;
